xmtp_id/src/scw_verifier/remote_signature_verifier.rs

```diff
@@ -1,31 +1,31 @@
 use super::{SmartContractSignatureVerifier, ValidationResponse, VerifierError};
 use crate::associations::AccountId;
-use async_trait::async_trait;
 use ethers::types::{BlockNumber, Bytes};
-use std::sync::Arc;
-use tokio::sync::Mutex;
-use tonic::transport::Channel;
 
-use xmtp_proto::xmtp::identity::api::v1::{
-    identity_api_client::IdentityApiClient, VerifySmartContractWalletSignatureRequestSignature,
-    VerifySmartContractWalletSignaturesRequest, VerifySmartContractWalletSignaturesResponse,
+use xmtp_proto::{
+    api_client::trait_impls::XmtpApi,
+    xmtp::identity::api::v1::{
+        VerifySmartContractWalletSignatureRequestSignature,
+        VerifySmartContractWalletSignaturesRequest, VerifySmartContractWalletSignaturesResponse,
+    },
 };
 
-#[derive(Clone)]
-pub struct RemoteSignatureVerifier {
-    identity_client: Arc<Mutex<IdentityApiClient<Channel>>>,
+pub struct RemoteSignatureVerifier<C> {
+    identity_client: C,
 }
 
-impl RemoteSignatureVerifier {
-    pub fn new(identity_client: IdentityApiClient<Channel>) -> Self {
-        Self {
-            identity_client: Arc::new(Mutex::new(identity_client)),
-        }
+impl<C> RemoteSignatureVerifier<C> {
+    pub fn new(identity_client: C) -> Self {
+        Self { identity_client }
     }
 }
 
-#[async_trait]
-impl SmartContractSignatureVerifier for RemoteSignatureVerifier {
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl<C> SmartContractSignatureVerifier for RemoteSignatureVerifier<C>
+where
+    C: XmtpApi,
```

Comment:
We want this to be `XmtpIdentityClient`, the remote verifier doesn't need a copy of the whole api, just the identity part.